### PR TITLE
Exporting utils types to make it accessible for consumers

### PR DESCRIPTION
**Pull request by govind-srinidhi**
Relates to #401 

## 📝 Description

Exported util types so that the consumers can make use of it if needed

## 💥 Is this a breaking change (Yes/No):

- [x] No
- [ ] Yes (please describe the impact and migration path for existing Lume users)

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/Adyen/lume/blob/main/CONTRIBUTING.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/lib/src/types/utils.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+export type Color = Colors | DivergentColors | LegacyColors | OtherColors;
+
+export type Margins =
+  | 'auto'
+  | {
+      top?: number | 'auto';
+      right?: number | 'auto';
+      bottom?: number | 'auto';
+      left?: number | 'auto';
```
Comment (govind-srinidhi) [inline packages/lib/src/types/utils.ts:113]: Have made all the margin directions optional. 
Consumer should be allowed to pass few or all of them without typing issues.

However if consumer decides to pass nothing or an empty JSON, we set it to default margin.

File: packages/lib/src/types/dataset.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
-import { Color } from '@/utils/constants';
+import type { Color } from '@/types/utils';
 
-export type ChartType = 'grouped-bar' | 'line' | 'single-bar' | 'stacked-bar';
+export type BarLineChartType =
```
[the remark this answers is not in the document]
Comment (joao-m-santos) [reply packages/lib/src/types/dataset.ts:3]: Can you make it a sub section of the original `ChartType`?  (with `Exclude<ChartType, 'sparkline or wtv'>`)

File: packages/lib/src/components/charts/lume-bar-chart/lume-bar-chart.spec.ts [hunk 3]
```diff
@@ -9,7 +9,8 @@ import {
   yScale,
 } from '@test/unit/mock-data';
 import { flushPromises } from '@test/unit/flush-promises';
-import { CHART_TYPES, ChartType } from '@/utils/constants';
+import { CHART_TYPES } from '@/utils/constants';
+import type { ChartType } from '@/types/utils';
```
Comment (joao-m-santos) [inline packages/lib/src/components/charts/lume-bar-chart/lume-bar-chart.spec.ts:13]: We should probably consolidate these types/enums. For example, I don't think we need `CHART_TYPES` honestly - only being used in test files

File: packages/lib/src/types/dataset.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (govind-srinidhi) [reply packages/lib/src/types/dataset.ts:3]: Yeah, we can!

The problem with that approach is:
Original chart type has `bar` | `line` | `alluvial` | `sparkline`
But this chart type include: `'grouped-bar' | 'line' | 'single-bar' | 'stacked-bar'`

So we need to exclude three chart types from original and add 3 flavours of bar chart. 
`line` is the only common thing between these two types.

Maybe better to leave these two things separate 🤔 

What do you feel?

File: packages/lib/src/components/charts/lume-bar-chart/lume-bar-chart.spec.ts (hunk 3 above)
Comment (govind-srinidhi) [reply packages/lib/src/components/charts/lume-bar-chart/lume-bar-chart.spec.ts]: Updated ;)

File: packages/lib/src/types/dataset.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (joao-m-santos) [reply packages/lib/src/types/dataset.ts:3]: I think we should make a generic one with all options `bar, grouped-bar, line, sparkline, etc.` and then a subsets of that one as needed

**Review by joao-m-santos — APPROVED** [review]

**Pull request merged**